crates/test-utils/src/lib.rs

```diff
@@ -1,46 +1,54 @@
-use std::sync::atomic::{AtomicUsize, Ordering};
-
 use arbitrary::{Arbitrary, Unstructured};
-use rand::{rngs::OsRng, RngCore};
+use rand::{rngs::OsRng, CryptoRng, RngCore};
 
 pub mod bitcoin;
 pub mod bridge;
 pub mod evm_ee;
 pub mod l2;
 
-const ARB_GEN_LEN: usize = 1 << 24; // 16 MiB
+// Smaller buffer size as compared to 2^24
+const ARB_GEN_LEN: usize = 128;
 
 pub struct ArbitraryGenerator {
-    buf: Vec<u8>,
-    off: AtomicUsize,
+    buf: Vec<u8>, // Persistent buffer
 }
-
 impl Default for ArbitraryGenerator {
     fn default() -> Self {
         Self::new()
     }
 }
-
 impl ArbitraryGenerator {
     pub fn new() -> Self {
-        Self::new_with_size(ARB_GEN_LEN)
+        ArbitraryGenerator {
+            buf: vec![0u8; ARB_GEN_LEN],
+        }
+    }
+    pub fn new_with_size(s: usize) -> Self {
+        ArbitraryGenerator { buf: vec![0u8; s] }
     }
 
-    pub fn new_with_size(n: usize) -> Self {
-        let mut buf = vec![0; n];
-        OsRng.fill_bytes(&mut buf); // 128 wasn't enough
-        let off = AtomicUsize::new(0);
-        ArbitraryGenerator { buf, off }
+    /// Legacy interface: no arguments
```

Comment:
Is this deprecated?
Maybe add a https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute ?

Or maybe add a note that this uses `OsRng` in the docstrings.

Reply:
It's not deprecated actually, I updated the docs which provides context.

Reply:
Agree